fetch.bs

```diff
@@ -5343,10 +5402,22 @@ method, must run these steps:
  {{Headers}} object whose <a for=Headers>guard</a> is
  "<code>immutable</code>".
 
+ <li>If <var>requestObject</var>'s <a for=Request>signal</a> is not null and its
+ <a for=AbortSignal>aborted flag</a> is set, reject <var>p</var> with an
+ "<code><a exception>AbortError</a></code>" {{DOMException}}, and return <var>p</var>.
+
+ <li><p>Otherwise, if <var>requestObject</var>'s <a for=Request>signal</a> is not null, add the
+ following steps to <var>requestObject</var>'s <a for=Request>signal</a>'s
+ <a for=AbortSignal>abort algorithms</a>:
+
+ <ol>
+  <li><p><a lt=terminated for=fetch>Terminate</a> <var>ongoingFetch</var> with reason <i>fatal</i>.
+ </ol>
+
  <li>
   <p>Run the following <a>in parallel</a>:
 
-  <p><a for=/>Fetch</a> <var>request</var>.
+  <p><a for=/>Fetch</a> <var>request</var> and let <var>ongoingFetch</var> be the ongoing fetch.
```

Comment:
Can you set a flag that is shared between main thread and the in parallel steps?  Then you can have various "cancellation points" where you check that flag.  Somewhat similar to how the implementation is done in places.  Of course, cancelling the network stack doesn't work that way, but maybe you could poll the flag there.